types/seat/wlr_seat_touch.c

```diff
@@ -109,7 +109,8 @@ static void touch_point_handle_surface_destroy(struct wl_listener *listener,
 		void *data) {
 	struct wlr_touch_point *point =
 		wl_container_of(listener, point, surface_destroy);
-	touch_point_destroy(point);
+	// Touch point itself is destroyed on up event
+	point->surface = NULL;
```

Comment:
Need to remove the surface destroy listener (and `wl_list_init` it so that `touch_point_destroy` still works)

Reply:
Updated. Is that really necessary btw, `touch_point_destroy` would handle it.

Reply:
Seems like in this particular case it would be okay because the surface destroy handler doesn't try to access `point->surface`. It's a little bit fragile to rely on that though. Performing surface-related cleanup steps when unsetting the surface prevents bad surprises.

Reply:
thanks for double checking, at least my analysis was correct but let's better be consistent here. thanks!